src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueSupplyAdaptor.sol

```diff
@@ -0,0 +1,254 @@
+        ACCOUNT_FOR_INTEREST = _accountForInterest;
+        morphoBlue = IMorpho(_morphoBlue);
+    }
+
+    // ============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Morpho Blue Supply Adaptor V 0.1"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+    /**
+     * @notice Allows user, if Cellar has a MBSupplyAdaptorPosition as its holding position, to deposit into MB markets.
+     * @dev Cellar must approve Morpho Blue to spend its assets, then call deposit to lend its assets.
+     * @param assets the amount of assets to lend on Morpho Blue
+     * @param adaptorData adaptor data containing the abi encoded Morpho Blue market Id
+     * @dev configurationData is NOT used
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        // Deposit assets to Morpho Blue.
+        Id id = abi.decode(adaptorData, (Id));
+        _validateMBMarket(id);
+
+        MarketParams memory market = morphoBlue.idToMarketParams(id);
+        // (address _loanToken, , , ) = morphoBlue.idToMarketParams(id); // See IMorpho for `idToMarketParams` and uncomment this if we go with the conventional IMorphoBlue interface function
+        ERC20 loanToken = ERC20(market.loanToken);
+        loanToken.safeApprove(address(morphoBlue), assets);
+        _deposit(market, assets, address(this));
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(loanToken, address(morphoBlue));
+    }
+
+    /**
+     * @notice Cellars must withdraw from Morpho Blue lending market, then transfer assets to receiver.
+     * @dev Important to verify that external receivers are allowed if receiver is not Cellar address.
+     * @param assets the amount of assets to withdraw from Morpho Blue lending market
+     * @param receiver the address to send withdrawn assets to
+     * @param adaptorData adaptor data containing the abi encoded Id
+     * @dev configurationData is NOT used
+     */
+    function withdraw(uint256 assets, address receiver, bytes memory adaptorData, bytes memory) public override {
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        // Withdraw assets from Morpho Blue.
+        Id id = abi.decode(adaptorData, (Id));
+        _validateMBMarket(id);
+        MarketParams memory market = morphoBlue.idToMarketParams(id);
+        _withdraw(market, assets, address(this));
```

Comment:
I think `address(this)` should be `receiver`, so that the assets are transferred to the user, as opposed to the Cellar.

Reply:
Agreed. I've made the change!